### DISCUSSION TITLE
clone problem

### DISCUSSION THREAD

**Discussion by Ahmed-Alhashmi** [community/community#68467] [New to GitHub]
i have 2 accounts in GitHub & i have projects so i wanna use Each project has a different clone to a different account on the same device , so how i do that 

Example : 

project name : angular_corse == > GitHub account : X1 

project name : .NET_course ==> GitHub account : X2 

so how to clone it to different  repository & accounts on the same device

**Comment by jsimecek** [comment]
Github gets configured on a project basis, so you can push each project to a different account or even one project to both accounts. 

Go to angular_corse folder on your device and run these commands:
`git init` - initialises git for that folder
`git add .` - stash everything locally 
`git commit -m "first commit"` - commit stashed to your local git (will be in the ".git" folder `git init` created)
`git branch -M main` - set your default branch to be "main"
`git remote add origin <YOUR_GITHUB_REPOSITORY_URL>` - connect project git to one of your accounts
`git push -u origin main` - publish to that accounts repository

Then do that the same for .NET_course folder while changing the Github repository url for the other account and you are done.

If you want to have the same project on both Github accounts, just connect the folder with
`git remote add <NAME_FOR_OTHER_ACCOUNT> <YOUR_GITHUB_REPOSITORY_URL>`
and publish with `git push <NAME_FOR_OTHER_ACCOUNT> main`

Don't forget to create an SSH key on your device and add it (public key) to both Github accounts, so that you can access publish to them.

**Comment by sreshtech** [comment]
![first](https://github.com/community/community/assets/121309862/d86ca50b-5f28-4dfa-b6fe-28d54c1288bb)
![sec](https://github.com/community/community/assets/121309862/5c89bf17-5c01-4e27-9529-3e35f9c7911c)
![3rd](https://github.com/community/community/assets/121309862/2ff167c4-c31d-4f30-9aa8-4bdbcd34e8bc)

![4th](https://github.com/community/community/assets/121309862/97e9851e-6997-4d9e-bb61-ab263d92a3a3)
make sure you're login to your github account on vscode

**Comment by 1amkaiz3n** [comment]
To clone repositories from different GitHub accounts on the same device, you'll need to ensure you have set up your local Git configuration properly. Here's a step-by-step guide:

1.Set Up SSH Keys for Each GitHub Account:

Generate SSH keys for each GitHub account following GitHub's guide: [Generating a new SSH key and adding it to the ssh-agent](https://docs.github.com/en/authentication/connecting-to-github/adding-a-new-ssh-key-to-your-ssh-agent).
Add the SSH key to the respective GitHub account.
2.Update Your SSH Config:

Create or edit your SSH config file (**~/.ssh/config**). If it doesn't exist, create it.
Add the following configuration for each GitHub account:
```
# For GitHub account X1
Host github.com-X1
  HostName github.com
  User git
  IdentityFile ~/.ssh/your_ssh_key_for_X1

# For GitHub account X2
Host github.com-X2
  HostName github.com
  User git
  IdentityFile ~/.ssh/your_ssh_key_for_X2
 
```

3.Clone Repositories Using Custom Hosts:

Clone the repositories using the custom hosts you defined in the SSH config.
```
# For repository in GitHub account X1
git clone git@github.com-X1:username/repo_name.git

# For repository in GitHub account X2
git clone git@github.com-X2:username/repo_name.git
 
```


Replace **your_ssh_key_for_X1** and **your_ssh_key_for_X2** with the respective SSH keys for each GitHub account.

With this setup, you'll be able to clone and work with repositories from different GitHub accounts on the same device without conflicts. Just make sure to use the correct custom host (**github.com-X1** or **github.com-X2**) when cloning or interacting with each repository.
